core/tests/puppeteer-acceptance-tests/user-utilities/logged-in-users-utils.js

```diff
@@ -0,0 +1,691 @@
+   */
+  async clickLearnMoreAboutOppiaButtonInAboutFoundation() {
+    await this.clickOn(learnMoreAboutOppiaButton);
+    const newTab = await this.browserObject.waitForTarget(
+      target => target.url() === aboutUrl
+    );
+    if (newTab.url() !== aboutUrl) {
+      throw new Error (
+        'The Learn More About Oppia button does not open the About page!');
+    } else {
+      showMessage(
+        'The Learn More About Oppia button opens the About page.');
+      const newPage = await newTab.page();
+      await newPage.close();
+    }
+  }
+
+  /**
+   * Function to click the Become A Volunteer button
+   * in the About Foundation page and check if it opens the Volunteer page.
+   */
+  async clickBecomeAVolunteerButtonInAboutFoundation() {
+    await this.clickOn(becomeAVolunteerButton);
+    const newTab = await this.browserObject.waitForTarget(
+      target => target.url() === volunteerUrl
+    );
+    if (newTab.url() !== volunteerUrl) {
+      throw new Error (
+        'The Become A Volunteer button does not open the Volunteer page!');
+    } else {
+      showMessage(
+        'The Become A Volunteer button opens the Volunteer page.');
+    }
+  }
+
+  /**
+   * Function for click the Consider Becoming A Partner Today! button
+   * in the About Foundation page and check if it opens the Partnerships page.
+   */
+  async clickConsiderBecomingAPartnerTodayButtonInAboutFoundation() {
+    await this.page.waitForSelector(sectionSixPart1);
+    const buttonText = await this.page.$eval(
+      sectionSixPart1,
+      element => element.getElementsByTagName('a')[0].textContent
+    );
+    if (buttonText !== 'Consider becoming a partner today!') {
+      throw new Error (
+        'The Consider becoming a partner today! button does not exist!');
+    } else {
+      await this.page.$eval(
+        sectionSixPart1,
+        element => element.getElementsByTagName('a')[0].click()
+      );
+      if (this.page.url() !== partnershipsUrl) {
+        throw new Error (
+          'The Consider becoming a partner today! button does not open ' +
+          'the Partnerships page!');
+      } else {
+        showMessage(
+          'The Consider becoming a partner today! button opens ' +
+          'the Partnerships page.');
+      }
+    }
+  }
+
+  /**
+   * Function for click the Join our large volunteer community! button
+   * in the About Foundation page and check if it opens the Volunteer page.
+   */
+  async clickJoinOurLargeVolunteerCommunityButtonInAboutFoundation() {
+    await this.page.waitForSelector(sectionSixPart2);
+    const buttonText = await this.page.$eval(
+      sectionSixPart2,
+      element => element.getElementsByTagName('a')[0].textContent
+    );
+    if (buttonText !== 'Join our large volunteer community!') {
+      throw new Error (
+        'The Join our large volunteer community! button does not exist!');
+    } else {
+      await this.page.$eval(
```

Comment:
I think for this and similar constructs, you can pull out the stuff in the "else" to the main part of the code, and drop the else block. The code will terminate anyway if it hits the if block, and less nesting makes things easier to read. You can use a similar pattern below.